adapt/engine.py

```diff
@@ -121,8 +164,18 @@ def register_intent_parser(self, intent_parser):
 
         :raises ValueError on invalid intent
         """
-        if hasattr(intent_parser, 'validate') and callable(intent_parser.validate):
-            self.intent_parsers.append(intent_parser)
+        if hasattr(
+                intent_parser,
+                'validate') and callable(
+                intent_parser.validate):
+            missing_entities = self.trie.checkForMissingEntites(
```

Comment:
This is a potentially breaking change in behavior. I would recommend adding an optional kwarg to this method, strict=False (or something like that) that optionally validates that vocab and parser are setup in this order.

There's a valid use case where vocabulary comes in after intent registration (async and dynamically from a remote source, like polling a 3P web service) where this would need to be disabled. I recommend off by default.